exchanges/stream/websocket_test.go

```diff
@@ -92,416 +97,343 @@ type dodgyConnection struct {
 
 // override websocket connection method to produce a wicked terrible error
 func (d *dodgyConnection) Shutdown() error {
-	return errors.New("cannot shutdown due to some dastardly reason")
+	return fmt.Errorf("%w: %w", errCannotShutdown, errDastardlyReason)
 }
 
 // override websocket connection method to produce a wicked terrible error
 func (d *dodgyConnection) Connect() error {
-	return errors.New("cannot connect due to some dastardly reason")
+	return fmt.Errorf("cannot connect: %w", errDastardlyReason)
+}
+
+func TestMain(_ *testing.M) {
+	// Change trafficCheckInterval for TestTrafficMonitorTimeout before parallel tests to avoid racing
+	trafficCheckInterval = 50 * time.Millisecond
 }
```

Comment:
Or the tests don't actually run
```suggestion
func TestMain(t *testing.M) {
	// Change trafficCheckInterval for TestTrafficMonitorTimeout before parallel tests to avoid racing
	trafficCheckInterval = 50 * time.Millisecond
	os.Exit(t.Run())
}
```

Reply:
Bwahahaha! Well. Fixed the race, didn't it?

FWIW, `m` should be called `m`.
Not like I'm on firm ground to be slinging corrections though!

Fixed 56b62692 and 7dd00dc9